Fix/update workflow

tirkarp opened the pull request:
#8 Updated dependencies and made workflow similar to `studiokit-net-js`.

gavin-brown commented:
Test comment

gavin-brown commented:
Test

gavin-brown commented:
Test

gavin-brown commented:
When building the `lib` folder, the correct files are getting built, in that there's a .js file and a corresponding d.ts file. However, these files are inside a `src` folder which is inside the `lib` folder, inside of being at the root of `lib`. There's also an extra `tests` folder inside the `lib` folder

studiokit-net-js  
<img width="419" alt="Screen Shot 2020-06-04 at 2 45 24 PM" src="https://user-images.githubusercontent.com/25330826/83798315-165e9000-a672-11ea-887d-a561dfe739df.png">



This branch  

<img width="616" alt="Screen Shot 2020-06-04 at 2 45 16 PM" src="https://user-images.githubusercontent.com/25330826/83798352-1eb6cb00-a672-11ea-904f-5a00ef5e85f6.png">

That it's building the right files is fantastic though!

tirkarp commented:
> When building the `lib` folder, the correct files are getting built, in that there's a .js file and a corresponding d.ts file. However, these files are inside a `src` folder which is inside the `lib` folder, inside of being at the root of `lib`. There's also an extra `tests` folder inside the `lib` folder
> 
> studiokit-net-js
> <img alt="Screen Shot 2020-06-04 at 2 45 24 PM" width="419" src="https://user-images.githubusercontent.com/25330826/83798315-165e9000-a672-11ea-887d-a561dfe739df.png">
> 
> This branch
> 
> <img alt="Screen Shot 2020-06-04 at 2 45 16 PM" width="616" src="https://user-images.githubusercontent.com/25330826/83798352-1eb6cb00-a672-11ea-904f-5a00ef5e85f6.png">
> 
> That it's building the right files is fantastic though!

Fixed.

builtbywill commented on index.js:
```diff
@@ -14,8 +14,8 @@
 
-var CaliperService = require('./lib/caliper-service');
-var BrowserCaliperService = require('./lib/browser-caliper-service');
+var BrowserCaliperService = require('./src/browser-caliper-service');
```
@tirkarp we actually probably want to move this index entry file into the src folder, so it is included in the compiled output

tirkarp replied:
Moved.

builtbywill commented:
@tirkarp when i check out and run `yarn` i am setting this change in `yarn.lock`

![image](https://user-images.githubusercontent.com/689209/83804461-28ddc700-a67c-11ea-9e61-bd7a1ea869c1.png)

tirkarp commented:
> @tirkarp when i check out and run `yarn` i am seeing this change in `yarn.lock`

also fixed. yarn problem 😅

gavin-brown commented:
I got this error when running tests, which I believe was due to not using a built version. This was a user error on my part and not a problem with the library 

```
Cannot find module 'studiokit-caliper-js' from 'caliper.js'

    However, Jest was able to find:
    	'../../utils/caliper.d.ts'
    However, Jest was able to find:
    	'sagas/rootSaga.d.ts'
    	'sagas/rootSaga.js'

    You might want to include a file extension in your import, or update your 'moduleFileExtensions', which is currently ['web.js', 'js', 'web.ts', 'ts', 'web.tsx', 'tsx', 'json', 'web.jsx', 'jsx', 'node'].

    See https://jestjs.io/docs/en/configuration#modulefileextensions-array-string

    However, Jest was able to find:
    	'redux/configureStore.d.ts'
    	'redux/configureStore.js'

    You might want to include a file extension in your import, or update your 'moduleFileExtensions', which is currently ['web.js', 'js', 'web.ts', 'ts', 'web.tsx', 'tsx', 'json', 'web.jsx', 'jsx', 'node'].

    See https://jestjs.io/docs/en/configuration#modulefileextensions-array-string

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:259:17)
      at Object.<anonymous> (node_modules/studiokit-scaffolding-js/lib/utils/caliper.js:6:46
```

After creating a built version and using it in another project (https://gitlab.com/purdue-informatics/studiokit/studiokit-scaffolding-js/-/commits/feature/caliper-update-built) and attempting to run `yarn build`, I get this error. I'm not sure if this error is due to how I'm using it or an issue with the library

```
node_modules/studiokit-caliper-js/lib/browser-caliper-service.d.ts:5:16 - error TS1005: '(' expected.

5     constructor: typeof BrowserCaliperService;
                 ~

Found 1 error.
```

gavin-brown commented:
@tirkarp I think the issue above is due to a mismatch in Typescript versions https://forum.ionicframework.com/t/getting-error-ts1005-expected-what-does-it-mean/51106

This library is using 3.9.3 and scaffolding is using 3.7.5. Our other projects are also using 3.7.5. Just to be safe, we probably want this library to use 3.7.5 as well

tirkarp commented:
> After creating a built version and using it in another project (https://gitlab.com/purdue-informatics/studiokit/studiokit-scaffolding-js/-/commits/feature/caliper-update-built) and attempting to run `yarn build`, I get this error. I'm not sure if this error is due to how I'm using it or an issue with the library
> 
> ```
> node_modules/studiokit-caliper-js/lib/browser-caliper-service.d.ts:5:16 - error TS1005: '(' expected.
> 
> 5     constructor: typeof BrowserCaliperService;
>                  ~
> 
> Found 1 error.
> ```

Cloned the branch and ran `yarn build` on scaffolding, and it worked fine for me. Did you `yarn` it before you build?

gavin-brown commented:
> Cloned the branch and ran yarn build on scaffolding, and it worked fine for me. Did you yarn it before you build?

I'll try it again. We might still want the Typescript versions to match but not completely sure what the implications are there

gavin-brown commented:
Because other libraries are surely using different Typescript versions

tirkarp commented:
And now I'm getting this when using the new scaffolding with Circuit and Variate 😓 

`../studiokit-scaffolding-js/node_modules/react-overlays/esm/Dropdown.js
Attempted import error: 'useUncontrolledProp' is not exported from 'uncontrollable'.`

gavin-brown commented:
_With both Typescript 3.9.3 and 3.7.5_
I ran `yarn cache clean` and deleted all my `node_modules`

- Using the `fix/update-workflow-built` branch, I got the `error TS1005` message
- Using the non-built branch, `yarn build` worked

But, I do believe we need to be using the built version when using this library. Part of the step of publishing to NPM is doing `yarn build`

tirkarp commented:
Ah. Finally got the error reproduced.

gavin-brown commented:
I was hoping the Typescript source code might have more details about what the TS1005 error is but unfortunately it does not https://github.com/microsoft/TypeScript/blob/v3.7.5/src/compiler/diagnosticMessages.json#L10

tirkarp commented:
Fixed, I think. It might have been a Babel issue where it fails to transpile old JS code into TS, so I changed that file to use ES6 syntax. @gavin-brown can you confirm that it works on your machine?

gavin-brown commented:
With that update, scaffolding-js built successfully when using the built version of this library! And then the tests passed in Circuit. The next step will be to deploy Circuit to dev and test there

gavin-brown commented:
It took me a while but I finally got this going on Circuit dev. Once I did, I got this error

![image](https://user-images.githubusercontent.com/25330826/84934316-28debd80-b0a5-11ea-83b4-303a9860b192.png)

That could be an issue with how studiokit-scaffolding-js is using this library. It's currently being used like this: `service = new StudioKit.BrowserCaliperService(options)`

With the updates, it might need initialized differently?

gavin-brown commented:
Yep, usage in scaffolding needs updated. It's not an issue with this library

https://github.com/purdue-tlt/studiokit-caliper-js/pull/9#issuecomment-645541070

builtbywill commented on package.json:
```diff
@@ -1,66 +1,134 @@
+			"studiokit"
+		],
+		"rules": {
+			"prettier/prettier": [
```
@tirkarp we probably want to ditch these prettier rule overrides, since our shared rules should be coming from extending the studiokit config

also when I run `yarn fix` I am seeing a number of errors, mostly related to the no-unused-vars rule

tirkarp replied:
Removed. Those eslint errors are fixed in `fix/eslint` branch.
